File: Backend/AIBackend/Dominio/Model/PesquisaModel.cs
namespace AIBackend.Dominio.Model
{
    public class PesquisaModel
    {
        public int Id { get; set; }
        public string TextoPesquisa { get; set; }
        public int IdTipoPesquisa { get; set; }

        public static explicit operator PesquisaModel(PesquisarCommand source)
        {
            return new PesquisaModel()
            {
                IdTipoPesquisa = source.TipoPesquisa,
                TextoPesquisa = source.Pesquisa
            };
        }
    }
}

File: ai_ui/lib/models/pesquisaModel.dart
class PesquisaModel {
  int id;
  String textoPesquisa;
  int idTipoPesquisa;

  PesquisaModel(this.id, this.textoPesquisa, this.idTipoPesquisa);

  factory PesquisaModel.fromJson(Map<dynamic, dynamic> json) {
    return PesquisaModel(
      json['id'] as int,
      json['textoPesquisa'] as String,
      json['idTipoPesquisa'] as int
    );
  }
}